agtern/common/api.py

```diff
@@ -1,11 +1,44 @@
-"""Pre-MVP: This file calls functions in agtern.backend.server to retrieve data.
-Post-MVP: This file will provide functions that will query the AgTern web server."""
+import requests
+from requests.adapters import HTTPAdapter, Retry
 
-from typing import List
+from .schemas import Internship, InternshipCreate
+from .logger import LOG
 
-from .models import Internship
-from ..server import get_all_internships
 
+class AgTernAPI:
+    """Wrapper for all communication to AgTern's services."""
+    env = "dev"
+    LOCALHOST_URL = "http://127.0.0.1:5000"
+    SERVER_URL = "http://our-public-agtern-api.com"
 
-def api_get_all_internships() -> List[Internship]:
-    return get_all_internships()
+    def __init__(self):
+        # TODO: make this env dynamic based off arguments
+        if self.env == "dev":
+            self.base_url = self.LOCALHOST_URL
+        else:
+            # NOTE: update this when we are cloud-hosted
+            self.base_url = self.SERVER_URL
+        self.session = requests.Session()
+        retry = Retry(total=5, backoff_factor=1)
+        self.session.mount("http://", HTTPAdapter(max_retries=retry))
+
+    def get_all_internships(self):
+        """Retrieve all internships."""
+
+        LOG.info("Retrieving internships...")
+        resp = self.session.get(self.base_url + "/internships/")
+        if resp.ok:
+            data = resp.json()
+            return [Internship(**iship) for iship in data]
+        return []
+
+    def create_internship(self, internship: InternshipCreate):
+        """Create an internship."""
+
+        LOG.info("Creating internship...")
+        resp = self.session.post(self.base_url + "/internships/",
+                                 data=internship.json())
+        if resp.ok:
+            data = resp.json()
+            return Internship(**data)
```

Comment:
Since `Internship` is a Pydantic Model, it may raise `ValidationErrors`. Currently, if there is a validation error, the whole program crashes. Consider something like this:
```suggestion
            try:
                return Internship(**data)
            except ValidationError as errors:
                LOG.error("Unable to create internship!")
                LOG.error(errors)
```